server.go

```diff
@@ -109,6 +109,8 @@ type server struct {
 	nat                  NAT
 	db                   database.Db
 	timeSource           blockchain.MedianTimeSource
+	services             wire.ServiceFlag
+	servicesMtx          sync.Mutex
```

Comment:
This mutex shouldn't be needed.  The `services` field is set during the initial server creation before and it it's not changed after that.